src/test/lib/ApiTest.cpp

```diff
@@ -1152,19 +1152,48 @@ DummyStreamCallback(
     return QUIC_STATUS_SUCCESS;
 }
 
+
 struct CloseFromCallbackContext {
     uint16_t CloseCount;
     volatile uint16_t CurrentCount;
+    // assuming server/client have 1 connection
+    MsQuicStream* ServerStreams[10] = {};
+    uint8_t NumServerStream {0};
+    MsQuicStream* ClientStreams[10] = {};
+    uint8_t NumClientStream {0};
```

Comment:
I _believe_ this is all mostly unnecessary. When you use `CleanUpAutoDelete` for a stream, it will automatically `Close` when the stream is shutdown.

Reply:
No, streams won't be deleted automatically until QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE arrives.

I saw "abort" happened at the end of the test when CloseCount is 5-7. The Server close his Connection before closing Stream.

Reply:
hmm, code flow is something like this. This does Open/Start/Send, but I don't see any stream event against the problematic stream.
```c++
MsQuicStream* Stream = new(std::nothrow) MsQuicStream(*Conn, QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL, CleanUpAutoDelete, StreamCallback, Context);
Stream->Send(&Ctx->LBuffer, 1, QUIC_SEND_FLAG_START | QUIC_SEND_FLAG_FIN);
Conn->Close();
```